### ISSUE TITLE
Data Augmentation during Training?

### ISSUE THREAD

**Issue by Elena-ssq** [lmb-freiburg/dispnet-flownet-docker#10]
Hi,

I've encountered a strange problem working with the net and I wonder if you applied data augmentation while training a disparity network?

Here I describe the details in case it may help:

The ground-truth and disparity estimation seems fine during traing, and the loss between then is acceptable. However, when I use the trained net to estimate an image pair, the results are much smaller than they should be.

I also checked the training curve and use image pairs from training set to make sure the problem is not caused by overfitting.

So at present, my best guess is that the data augmentation destroys the original relation between stereo image pairs and leads to wrong estimations. By the way, I did follow the data augmentation layer settings provided.

Please help, Tkx.

**Comment by nikolausmayer** [comment]
Hi,

did you use the conversion scripts in the `/tools` subfolder? I just checked and there's a bug in `convert_imageset_and_disparity.cpp` where line 158 should be:
<code>value = <b>32 * </b>cv_disp.at<float>(y, x);</code>

Without this, the estimates are likely too small by a factor of ca. 32.

(This was fixed in the direct-download version of DispNet, but apparently not in the codebase downloaded by the Dockerfile.)

**Comment by Elena-ssq** [comment]
Thanks for the note.

Yes, the file did miss the factor. I'll try again and that should work.

Still, I noticed that after removing data augmentation during training, the network seems to converge faster and smoother. So does the augmentation really help improving network performance other than potentially avoid overfitting?

**Comment by nikolausmayer** [comment]
> after removing data augmentation during training, the network seems to converge faster and smoother

I would say this is to be expected, but it's directly related to overfitting: by removing the augmentation, you are limiting the dataset variance and making it easier to overfit to it. The dataset should be large enough to teach the network proper disparity matching, but it cannot teach e.g. how to deal with "bright left image, dark right image"; for such effects we use augmentation.

**Comment by Elena-ssq** [comment]
In that case augmentation is in deed helpful.

Thanks for the help. The problem is well solved after correcting the cpp file.

**Comment by nikolausmayer** [comment]
Thanks, I've updated the archive which is downloaded during the build process.
